### PR TITLE
Allow comma separated list of container names in dependsOn elements

### PR DESCRIPTION
**Pull request by chonton**
### Description
Allow comma separated list of <dependsOn> elements.  (Similar to #558)
### Info

* Feature request: Allow something like:
```
<dependsOn>
   <container>mongo,identity</container>
</dependsOn>
```

### CODE REVIEW

File: src/main/java/io/fabric8/maven/docker/config/ImageConfiguration.java [hunk 1]
```diff
@@ -97,7 +97,7 @@ private void addVolumes(RunImageConfiguration runConfig, List<String> ret) {
 
     private void addLinks(RunImageConfiguration runConfig, List<String> ret) {
         // Custom networks can have circular links, no need to be considered for the starting order.
-        if (runConfig.getLinks() != null && !runConfig.getNetworkingConfig().isCustomNetwork()) {
```
Comment (rhuss) [inline src/main/java/io/fabric8/maven/docker/config/ImageConfiguration.java]: Are you sure that getLinks() is never null ?

File: src/main/java/io/fabric8/maven/docker/config/ImageConfiguration.java [hunk 2]
```diff
@@ -114,7 +114,7 @@ private void addContainerNetwork(RunImageConfiguration runConfig, List<String> r
 
     private void addDependsOn(RunImageConfiguration runConfig, List<String> ret) {
         // Only used in custom networks.
-        if (runConfig.getDependsOn() != null && runConfig.getNetworkingConfig().isCustomNetwork()) {
```
Comment (rhuss) [inline src/main/java/io/fabric8/maven/docker/config/ImageConfiguration.java]: why can't getDependsOn() be null ?

File: src/main/java/io/fabric8/maven/docker/config/ImageConfiguration.java (hunk 1 above)
Comment (chonton) [inline src/main/java/io/fabric8/maven/docker/config/ImageConfiguration.java]: I've added @Nonnull annotations to show the deduction chain.  EnvUtil.splitAtCommasAndTrim(Iterable<String> input) never returns null.

File: src/main/java/io/fabric8/maven/docker/config/ImageConfiguration.java (hunk 2 above)
Comment (chonton) [inline src/main/java/io/fabric8/maven/docker/config/ImageConfiguration.java]: likewise.

File: src/main/java/io/fabric8/maven/docker/config/ImageConfiguration.java (hunk 1 above)
Comment (rhuss) [inline src/main/java/io/fabric8/maven/docker/config/ImageConfiguration.java]: Ok, I'm settled then ;-)

**Pull request merged**